dexcom/fetch/runner.go

```diff
@@ -124,17 +124,16 @@ func (r *Runner) Run(ctx context.Context, tsk *task.Task) bool {
 	}
 
 	if !skipToAvoidDexcomBackup {
-		tsk.ClearError()
+		ResetTask(tsk)
 
 		if serverSessionToken, sErr := r.AuthClient().ServerSessionToken(); sErr != nil {
-			tsk.AppendError(errors.Wrap(sErr, "unable to get server session token"))
+			ErrorOrRetryTask(tsk, errors.Wrap(sErr, "unable to get server session token"))
 		} else {
 			ctx = auth.NewContextWithServerSessionToken(ctx, serverSessionToken)
-
 			if taskRunner, tErr := NewTaskRunner(r, tsk); tErr != nil {
-				tsk.AppendError(errors.Wrap(tErr, "unable to create task runner"))
+				ErrorOrRetryTask(tsk, errors.Wrap(tErr, "unable to create task runner"))
 			} else if tErr = taskRunner.Run(ctx); tErr != nil {
-				tsk.AppendError(errors.Wrap(tErr, "unable to run task runner"))
+				ErrorOrRetryTask(tsk, errors.Wrap(tErr, "unable to run task runner"))
```

Comment:
I think this is the only place where we should fail the task if retries are exceeded.